src/main/kotlin/org/rust/lang/core/psi/impl/RustStubbedItemImpl.kt

```diff
@@ -0,0 +1,26 @@
+package org.rust.lang.core.psi.impl
+
+import com.intellij.lang.ASTNode
+import com.intellij.psi.stubs.IStubElementType
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.RustItem
+import org.rust.lang.core.psi.RustNamedElement
+import org.rust.lang.core.psi.RustOuterAttr
+import org.rust.lang.core.psi.RustVis
+import org.rust.lang.core.stubs.RustItemStub
+
+open class RustStubbedItemImpl : RustStubbedNamedElementImpl<RustItemStub>, RustItem {
+
+    constructor(node: ASTNode) : super(node)
+
+    constructor(stub: RustItemStub, nodeType: IStubElementType<*, *>) : super(stub, nodeType)
+
+    override fun getOuterAttrList(): List<RustOuterAttr> =
+        PsiTreeUtil.getChildrenOfTypeAsList(this, RustOuterAttr::class.java)
```

Comment:
It should go like this: you need to check whether this element is backed by the stub-tree instead of AST by checking whether `getStub()` returns `null` or not, in the case we're in stub-tree, we'd _not_ touch any PSI related contents, otherwise we'd trigger parse implicitly.